### PR TITLE
Pbft fixes

### PR DESCRIPTION
**Pull request by JakubFornadel**
## Purpose

Few fixes:

- current pbft period is always chain_size + 1
-  removed two_t_plus_one value from db as we dont need it - we can use current one
- send proposed pbft block before propose vote

### CODE REVIEW

**Review by justinsnapp — APPROVED** [review]

File: libraries/core_libs/network/src/tarcap/stats/node_stats.cpp [hunk 1]
```diff
@@ -28,6 +28,14 @@ NodeStats::NodeStats(std::shared_ptr<PeersState> peers_state, std::shared_ptr<Pb
       packets_stats_(std::move(packets_stats)),
       thread_pool_(std::move(thread_pool)) {
   LOG_OBJECTS_CREATE("SUMMARY");
+
+  local_pbft_step_prev_interval_ = pbft_mgr_->getPbftStep();
```
Comment (JakubFornadel) [inline libraries/core_libs/network/src/tarcap/stats/node_stats.cpp:32]: @justinsnapp  you cant do this - some tests, e.g.
```
TEST_F(NetworkTest, transfer_block) {
  auto nw1 = std::make_unique<Network>(g_conf1);
  auto nw2 = std::make_unique<Network>(g_conf2);
```

create network without any managers so this segfaults. I am reverting this commit

Comment (justinsnapp) [reply libraries/core_libs/network/src/tarcap/stats/node_stats.cpp]: Ok problem was just including it in the constructor then at least without a check for pbft_mgr being initizalized.  We already call pbft_mgr in the logging function though so no reason to completely revert this commit @JakubFornadel

Comment (JakubFornadel) [reply libraries/core_libs/network/src/tarcap/stats/node_stats.cpp]: you can cherry pick it and fix it in your separate branch then. I wanted to continue with my changes and this broke my branch so I reverted it...

**Review by kstdl — APPROVED** [review]

**Pull request merged**